app/assets/javascripts/tinymce/plugins/alchemy_link/plugin.min.js

```diff
@@ -5,15 +5,29 @@ tinymce.PluginManager.add("alchemy_link", function (editor) {
 
   const openLinkDialog = () => {
     if (Alchemy.currentDialog()) return
-    const linkObject = {
-      node: getAnchor(editor.selection.getNode()),
-      bookmark: editor.selection.getBookmark(),
-      selection: editor.selection,
-      editor: editor
+
+    let link = {}
+    const anchor = getAnchor(editor.selection.getNode())
+    if (anchor) {
+      link = {
+        url: anchor.href,
+        title: anchor.title,
+        target: anchor.getAttribute("data-link-target"),
+        type: anchor.className
+      }
     }
-    const linkDialog = new Alchemy.LinkDialog(linkObject)
+    const linkDialog = new Alchemy.LinkDialog(link)
     editor.focus()
-    linkDialog.open()
+    linkDialog.open().then((link) => {
+      editor.execCommand("mceInsertLink", false, {
+        href: link.url,
+        class: link.type,
+        title: link.title,
+        "data-link-target": link.target,
+        target: link.target === "blank" ? "_blank" : undefined
```

Comment:
I think we must only set the target and `data-link-target` (which we can remove) only if `target` is set. Like this it is set to an empty string and the value `undefined` on anchor tags.

```js
const options = {
  href: link.url,
  class: link.type,
  title: link.title
}

if (link.target === "blank") {
  options.target = "_blank"
}
```